### PR TITLE
Revit_Toolkit-Issue261-ProfileExtraction

### PR DESCRIPTION
**Pull request by ZiolkowskiJakub**
### Issues addressed by this PR
Functionality to Load multiple families and place FamilySymbols

PR related to #261 

### Test files
Sample script and families for tests saved [here](https://burohappold.sharepoint.com/:f:/r/sites/BHoM/02_Current/12_Scripts/01_Test%20Scripts/Revit_Toolkit/Revit_Toolkit-Issue261-ProfileExtraction?csf=1&e=d0JR1k)

### Changelog
RevitFilePreview Push loads family
Revit_Toolkit Family PlacementType information added
Path property added to RevitFilePreview
RevitFilePreview to Revit_Toolkit Family method added
RevitFilePreview Paths method updated to pull unique path only

### CODE REVIEW

**Comment by michaldengusiak** [comment]
@ZiolkowskiJakub 
it does not build Revit for me, 
image

**Comment by ZiolkowskiJakub** [comment]
This is doe to changes has been appiled in following PR:
#304 

Please pull BHoM_Engine from master, Revit_Toolkit (this branch) and rebuild and try again.

Thanks.

File: Adapter_Revit_UI/Create/Create.cs [hunk 1]
```diff
@@ -260,28 +295,55 @@ private static void SetIdentifiers(IBHoMObject bHoMObject, Element element)
...
+                Parameter aParameter = null;
 
-            aParameter = element.get_Parameter(BuiltInParameter.ELEM_FAMILY_PARAM);
-            if (aParameter != null)
-                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.FamilyName, aParameter.AsValueString());
+                aParameter = element.get_Parameter(BuiltInParameter.ELEM_FAMILY_PARAM);
+                if (aParameter != null)
+                {
+                    string aValue = aParameter.AsValueString();
+                    if (string.IsNullOrEmpty(aValue))
```
Comment (IsakNaslundBh) [inline Adapter_Revit_UI/Create/Create.cs]: Check if this should be the opposite? Right now setting the values only if it is null or empty.

File: Adapter_Revit_UI/Create/Create.cs [hunk 2]
```diff
@@ -260,28 +295,55 @@ private static void SetIdentifiers(IBHoMObject bHoMObject, Element element)
...
-                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.FamilyTypeName, aParameter.AsValueString());
 
-            aParameter = element.get_Parameter(BuiltInParameter.ELEM_CATEGORY_PARAM);
-            if (aParameter != null)
-                SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.CategoryName, aParameter.AsValueString());
+                aParameter = element.get_Parameter(BuiltInParameter.ELEM_TYPE_PARAM);
+                if (aParameter != null)
+                {
+                    string aValue = aParameter.AsValueString();
+                    if (string.IsNullOrEmpty(aValue))
```
Comment (IsakNaslundBh) [inline Adapter_Revit_UI/Create/Create.cs]: as above

File: Adapter_Revit_UI/Create/Create.cs [hunk 3]
```diff
@@ -260,28 +295,55 @@ private static void SetIdentifiers(IBHoMObject bHoMObject, Element element)
...
+                        SetCustomData(bHoMObject, BH.Engine.Adapters.Revit.Convert.FamilyTypeName, aValue);
+                }
+
+
+                aParameter = element.get_Parameter(BuiltInParameter.ELEM_CATEGORY_PARAM);
+                if (aParameter != null)
+                {
+                    string aValue = aParameter.AsValueString();
+                    if (string.IsNullOrEmpty(aValue))
```
Comment (IsakNaslundBh) [inline Adapter_Revit_UI/Create/Create.cs]: as above

**Comment by IsakNaslundBh** [comment]
Script runs well for me, able to place beams multiple beams. Had problems with some of the types, but could be solved in separate issue/PR.

Can not place columns as they are two-level based and we do not have a method for that, but can also be sorted as separate issue. Issue for it raised here:

https://github.com/BHoM/Revit_Toolkit/issues/308

Will let Jakub have a look at the comments made in the code, and test that my previous scripts work, as soon as that is done, I am happy to approve.

@michaldengusiak Can you raise a separate issue for the detail items, as the process seem to be working for you as well for the general families?

**Comment by ZiolkowskiJakub** [comment]
> it works for standard families like MEP Equipment but does not work for detail items
> image

This was reported as separate issue here #309

**Pull request merged**